core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -251,6 +251,10 @@ class Partition(val topicPartition: TopicPartition,
   @volatile var leaderReplicaIdOpt: Option[Int] = None
   @volatile private[cluster] var isrState: IsrState = CommittedIsr(Set.empty)
   @volatile var assignmentState: AssignmentState = SimpleAssignmentState(Seq.empty)
+  @volatile private var logSlowReplicationBucketStartTime: Long = 0
+  @volatile private var logSlowReplicationBucketCounter: Int = 0
+  val logSlowReplicationBucketLengthMs: Long = 60000
+  val logSlowReplicationBucketMaxLogCount: Int = 1000
```

Comment:
This can still hit 16 QPS sustained (on average), so it might be good to dial it back to, say, 300? That would be 5 QPS, which seems more reasonable on a (presumably) stressed system, and it's also still providing a LOT of logging data. Is the higher quantity useful? For small bursts of latency you still get everything; you only lose logging data if it's hundreds per minute, in which case, how much value is associated with more?

Reply:
I think 5 QPS should be fine. I'll make the change.